added "Fail build if all/any scenarios failed" as an option

tommywo opened the pull request:
this is my fix for 
https://issues.jenkins-ci.org/browse/JENKINS-30081
screen:
<img width="942" alt="screen shot 2015-11-17 at 15 49 05" src="https://cloud.githubusercontent.com/assets/11270518/11214664/cf8866c6-8d42-11e5-9c7d-bf50f964f70b.png">

jtnord commented:
I'm not going to accept this on principal that it has a travis file...

jtnord commented:
Thank you for you contribution.
Fail if any and fail if all should not be 2 checkboxes as all implies any. So this should be a failed test dropdown which sets a behaviour (defaulting to mark unstable if any failures)
It should also be in an advanced section as I think most users would not want to change this.

tommywo commented:
I've changed two checkbox into a dropdown and added a test for build results. 
![screen shot 2015-11-18 at 10 55 17 1](https://cloud.githubusercontent.com/assets/11270518/11237948/f3a0e0c4-8de2-11e5-830c-573eba01c39d.png)
![screen shot 2015-11-18 at 10 55 28](https://cloud.githubusercontent.com/assets/11270518/11237956/fdb3743c-8de2-11e5-8330-713802f2c478.png)

but now one of tests failed on [jenkins](https://jenkins.ci.cloudbees.com/job/plugins/job/cucumber-testresult-plugin/77/testReport/) failed which is strage for me because it is passing when i run it. does it run differently when build on server?

jtnord commented on src/main/java/org/jenkinsci/plugins/cucumber/jsontestsupport/CucumberTestResultArchiver.java:
```diff
@@ -166,10 +179,15 @@ public boolean getIgnoreBadSteps(){
 		CHECKPOINT.report();
 
-		if (action.getResult().getTotalCount() == action.getResult().getFailCount()){
+		if (setResult.equals("failIfAll") && action.getResult().getTotalCount() == action.getResult().getFailCount()){
```
setResult may be `null`

jtnord commented:
the failure is as the `setResult` may be `null`.

I don't think these should be strings and treated like this but Objects that can have the logic inside them (which makes for easier testing and extensibility) and internationlization of the UI.  This may require a new ExtensionPoint to be implemented, or an enum (but enums are harder to evolve/extend)

tommywo commented:
Closing for now. I will submit a new pull request, if I have time to work on it.
